float32 power1
float32 power2